src/extensionFunctions.ts

```diff
@@ -599,4 +601,46 @@ export namespace extensionFunctions {
 			console.log('[' + msg + ']');
 		}
 	}
+
+	export async function validateCommandIDsInSelectedFile(didactUri: vscode.Uri) : Promise<void> {
+		if (didactUri) {
+			await vscode.commands.executeCommand(START_DIDACT_COMMAND, didactUri);
+		}
+		if (DidactWebviewPanel.currentPanel) {
+			const commands : any[] = extensionFunctions.gatherAllCommandsLinks();
+			if (commands && commands.length > 0) {
+				sendTextToOutputChannel(`Starting validation...`);
+				let allOk = true;
+				for(let command of commands) {
+					// validate all commands we found
+					const parsedUrl = url.parse(command, true);
+					const query = parsedUrl.query;
+					if (query.commandId) {
+						const commandId = utils.getValue(query.commandId);
+						if (commandId) {
+							const vsCommands : string[] = await vscode.commands.getCommands(true);
+							var filteredList : string[] = vsCommands.filter( function (command) {
+								return command === commandId;
+							});
+							if (filteredList.length === 1) {
+								// expected result - we found the command in the vscode command list
+							} else {
+								// unexpected result - let the user know
+								sendTextToOutputChannel(`--Missing Command ID ${commandId}`);
+								allOk = false;
+							}
+						}
+					}
```

Comment:
>     * adding a test would be useful

Yes, it would. But I don't have a good way to provide one. The feedback is all in the output right now. 

>     * when I read the goal of the feature, I was expecting to have validation inside the file. it will allow automatic feedback, put the marker where the command is invalid. Why did you choose to use a command and context menu? Easier to implement?
>     * each time I click on validate didact file from the toolbar menu, the preview is opening

I am not adding editor features here. That's way beyond the scope of Didact. 

We are parsing the HTML generated from the Markdown or AsciiDoc, which has to be done from the Didact window itself.

>     * it would be nice to put focus on the output log of Didact when calling the command as the result is visible only there. Otherwise, it looks alike nothing happened for the user

I'll see what we can do there.

>     * the validation against the saved file. I would expect to be launched to the file whcih is currently edited. Especially when using the editor toolbar menu

See earlier comments about doing it in the preview. 
 
>     * the command "validate didact file" is always available. If there is no Didact file opened in editor, it is silently doing nothing

Can we filter commands by "when"?

Reply:
/!\ this long commetn was placed on the wrong feedback. I replied to some parts of it in dedicated global comments.

Adding this message to be sure that the initial comment is still visible:

`I think it will be easier to read to extract a validateCommands method as it is already the 4th nested level. It will also avoid having to write a comment to precise it`